secure-storage/secure-storage-impl/src/main/java/com/duckduckgo/securestorage/impl/encryption/EncryptionHelper.kt

```diff
@@ -69,66 +54,39 @@ interface EncryptionHelper {
 
 @ContributesBinding(AppScope::class)
 class RealEncryptionHelper @Inject constructor() : EncryptionHelper {
-    private val cipher = Cipher.getInstance(TRANSFORMATION)
+    private val encryptionCipher = Cipher.getInstance(TRANSFORMATION)
```

Comment:
The root cause here was Cipher is not thread safe. Prior to this change:
- There are 4 functions that use the same cipher instance: 2 encrypt methods and 2 decrypt methods.
- Cipher maintains an internal state. Calling .init() resets the states.
- On the event that there's a race condition that one of the 4 functions are called simultaneously before the functions complete, then some issues could be encountered like overwriting encryption and decryption state.